pandas/core/indexes/base.py

```diff
@@ -1765,6 +1765,29 @@ def _validate_names(
 
         return new_names
 
+    def get_default_index_names(self, names=None, default=None) -> Sequence[str]:
+        from pandas.core.indexes.multi import MultiIndex
+
+        # if names is not None and not all(isinstance(name, str) for name in names):
```

Comment:
are there unit tests for all of these cases?

Reply:
i think there are not, pls add in the indexes test suite for both Index and MultiIndex which hit all cases here (some of which are too restrictive)

Reply:
Have add test case for index renames for single index and multi-index